pkg/kubelet/apis/kubeletconfig/v1beta1/types.go

```diff
@@ -39,6 +39,23 @@ const (
 	HairpinNone = "none"
 )
 
+// KubeletManagerMode denotes a mode in which internal manager (secret,
+// configmap) are running.
+type KubeletManagerMode string
+
+// Enum settings for different modes of kubelet managers.
+const (
+	// SimpleManagerMode is a mode in which kubelet fetches necessary objects
+	// directly from apiserver.
+	SimpleManagerMode KubeletManagerMode = "Simple"
+	// TTLCachingManagerMode is a mode in which kubelet uses ttl cache for
+	// object directly fetched from apiserver.
+	TTLCachingManagerMode KubeletManagerMode = "Caching"
+	// WatchingManagerMode is a mode in which kubelet uses watches to observe
+	// changed to objects that are in its interest.
+	WatchingManagerMode KubeletManagerMode = "Watching"
+)
+
```

Comment:
/cc @mtaufen FYI

Reply:
Thanks :). Comments below:

1. Is it necessary to expose this as a config? How risky do we think the change is? We already watch Pods, dynamic Kubelet config watches ConfigMaps, so watch is not unprecedented in the Kubelet.
2. If so, I think we can potentially improve the naming and structure here. Naming is hard :). See below.

With respect to "manager", I really like the advice in https://blog.codinghorror.com/i-shall-call-it-somethingmanager/

`Mode` is more general than it has to be, specifically this is a strategy for observing resource updates.

`ResourceNotifyStrategy` was my first guess of what to call the strategy type, but this is still a bit vague, IMO, because it is not clear who is being notified (is it the resource, or the Kubelet?), or why there is a notification.

`ResourceChangedEventStrategy` might be a bit better, since `ResourceChangedEvent` makes it clear what's going on. But `Event` is overloaded for another Kubernetes concept, so I don't like this either.

`ResourceChangeDetectionStrategy` manages to avoid overloading `Event`. This is my best guess so far, IMO. Still feels verbose, maybe there's a way to simplify it.

For a field name, maybe `ConfigMapAndSecretChangeDetectionStrategy`. It's a bit verbose, so if there's a standard term that clearly captures ConfigMaps and Secrets, and whatever other objects this field will cover in the future, consider using that. But be wary of making it too general.

Other things:
Should we allow the strategies for ConfigMaps and Secrets to be configured independently?

I think we can simply use the infinitive of the verb. Instead of `Caching` and `Watching`, we can just say `Cache` and `Watch`. `Simple` is a bit vague, would it be more accurate to call this strategy `Poll`?

Reply:
Also, structure-wise, is there a broader category for resource-consumption-lifecycles in the Kubelet, and should we invent a config substructure for that?

Reply:
I like: `ResourceChangeDetectionStrategy` - changed.

Regarding specific configmap and secrets:
- I don't have any idea for a common term for those two - so I left `ConfigMapAndSecretChangeDetectionStrategy`
- I don't want to split them, because we want to avoid diverging those two in my opinion

About specific infinitive verbs - yeah, that sounds fine. Also, I decided for "Fetch" instead of "Poll" - it's more accurate I think.

About the structure - I don't think we will have more about that, so introducing substructure doesn't make sense now in my opinion.